### PR TITLE
Add patch maintenance script for PostgreSQL

### PR DESCRIPTION
**Pull request by Kidswiss**
This PR adds security maintenance to the PostgreSQL by VSHN service. The maintenance will be run on a schedule.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation. *Don't forget to generate the CRD API!*
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Comment by Kidswiss** [comment]
As mentioned, the actual conversion from dayOfWeek and timeOfDay is a bit ugly, but works, if the environmnet-config alpha feature is enabled.

File: component/vshn_postgres.jsonnet [hunk 1]
```diff
@@ -638,6 +638,212 @@ local clusterRestoreConfig = {
...
+];
+
+local maintenanceJob = {
+  base: comp.KubeObject('batch/v1', 'CronJob') + {
+    spec+: {
+      forProvider+: {
+        manifest+: {
+          metadata+: {
+            namespace: pgParams.controlNamespace,
+          },
```
Comment (glrf) [inline component/vshn_postgres.jsonnet:804]: Nitpick: This is a bit misleading as we patch it anyways and it actually shouldn't run in the control namespace

**Review by glrf — APPROVED** [review]
LGTM. A bit hacky, but I don't have a cleaner solution

File: component/vshn_postgres.jsonnet (hunk 1 above)
Comment (Kidswiss) [reply component/vshn_postgres.jsonnet:804]: Whoops, you're right, that's probably a copy paste error. Good catch.

**Pull request merged**